com/bl/ttt/tictactoe.py

```diff
@@ -113,12 +121,11 @@ def is_computer_turn(self):
         return self.get_turn_player() == 'O'
 
     def is_game_over(self):
-        return self.has_won('X') or \
-               self.has_won('O') or \
-               self.is_draw()
+        if self._is_draw() or self._has_won(self.get_previous_turn_player()):
+            return True
 
     def inform_game_over(self):
-        if self.has_won(self.get_previous_turn_player()):
+        if self._has_won(self.get_previous_turn_player()):
```

Comment:
this get_previous_turn_player is a bit confusing.
when a player has taken their turn you really want all of the code to only be concerned with the current player who is taking their turn

Reply:
also this function really belongs in main.py

Reply:
get_previous_layer is removed and inform_game_over is refactored out of tictactoe.py and into main.py.